src/hook/useCustomRefForm.js

```diff
@@ -0,0 +1,44 @@
+import { useRef, useState } from "react";
+
+const useCustomRefForm = (initialInput, validate) => {
+  const inputRefs = useRef(initialInput);
+  const errorRefs = useRef({});
+  const [errors, setErrors] = useState({});
+
+  const validateInput = () => {
+    const validationErrors = validate(inputRefs.current);
+    errorRefs.current = validationErrors;
+    setErrors(validationErrors);
+    return Object.keys(validationErrors).length === 0;
+  };
+
+  const register = (name) => {
```

Comment:
read, what kind of things register return, try to implement common things

Reply:
it returns the 
reference of the element,
value of the elment
onChange function